lark/parsers/xearley.py

```diff
@@ -14,19 +14,25 @@
 Earley's power in parsing any CFG.
 """
 
+from typing import TYPE_CHECKING, Callable
 from collections import defaultdict
 
 from ..tree import Tree
 from ..exceptions import UnexpectedCharacters
 from ..lexer import Token
 from ..grammar import Terminal
 from .earley import Parser as BaseParser
-from .earley_forest import SymbolNode, TokenNode
+from .earley_forest import TokenNode
 
+if TYPE_CHECKING:
+    from ..common import LexerConf, ParserConf
 
 class Parser(BaseParser):
-    def __init__(self, lexer_conf, parser_conf, term_matcher, resolve_ambiguity=True, complete_lex = False, debug=False, tree_class=Tree):
-        BaseParser.__init__(self, lexer_conf, parser_conf, term_matcher, resolve_ambiguity, debug, tree_class)
+    def __init__(self, lexer_conf: 'LexerConf', parser_conf: 'ParserConf', term_matcher: Callable,
+                 resolve_ambiguity: bool=True, complete_lex: bool=False, debug: bool=False,
+                 tree_class: type=Tree, ordered_sets: bool=True):
```

Comment:
`tree_class` is Optional.

Reply:
See above, I don't think this is the case.

Reply:
I agree with MegaIng on this one. tree_class can't be None (except in `LarkOptions`, and even that might be a bit of a relic).